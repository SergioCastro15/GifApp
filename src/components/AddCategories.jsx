import { useState } from "react"
import { func } from "prop-types"

export const AddCategories = ({ onAddCategory }) => {

  const [inputValue, setInputValue] = useState('dragon ball')

  const onInputChange = (e) => {
    setInputValue(e.target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault()
    if( inputValue.trim().length <= 1) return;
    setInputValue('')
    onAddCategory(inputValue)
  }
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input 
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}

AddCategories.propTypes = {
  onAddCategory: func.isRequired,
}