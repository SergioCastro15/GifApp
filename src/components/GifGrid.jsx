import { string } from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem } from './GifItem'


export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category)
  
  return (
    <>
      <h3 data-testid="category">{category}</h3>
      { isLoading && <h2>Cargando ....</h2>}
      <div className='card-grid'>
        {
          images.map((image) => (
            <GifItem key={image.id} {...image} />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: string.isRequired,
}
