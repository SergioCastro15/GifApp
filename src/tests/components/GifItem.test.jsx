import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe(('<GifItem />'), () => {
  const title = 'title test';
  const url = 'https://one-punch.com/';

  test('debe hacer match con el snpashot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot();
  });

  test('debe mostrar la imagen con el URL y ALT indicado', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  });

  test('debe mostrar el titulo del component', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})