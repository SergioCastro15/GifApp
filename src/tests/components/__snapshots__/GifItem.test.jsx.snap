// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> debe hacer match con el snpashot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title test"
      src="https://one-punch.com/"
    />
    <p>
      title test
    </p>
  </div>
</div>
`;
